engine/common/follower/cache/distributor.go

```diff
@@ -0,0 +1,36 @@
+package cache
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+)
+
+type OnEntityEjected func(ejectedEntity flow.Entity)
+
+// HeroCacheDistributor wraps module.HeroCacheMetrics and allows subscribers to receive events
+// for ejected entries from cache.
+type HeroCacheDistributor struct {
+	module.HeroCacheMetrics
+	consumers []OnEntityEjected
+}
+
+var _ module.HeroCacheMetrics = (*HeroCacheDistributor)(nil)
+
+func NewDistributor(heroCacheMetrics module.HeroCacheMetrics) *HeroCacheDistributor {
+	return &HeroCacheDistributor{
+		HeroCacheMetrics: heroCacheMetrics,
+	}
+}
+
+func (d *HeroCacheDistributor) AddConsumer(consumer OnEntityEjected) {
+	d.consumers = append(d.consumers, consumer)
+}
```

Comment:
Should we protect this behind a lock? Or at least add a comment saying it isn't concurrency safe.

Reply:
I didn't add since it's not used in concurrent setup, I will add a comment

Reply:
https://github.com/onflow/flow-go/pull/3965/commits/7c9a3add0b12bacd969fa30ec8a73f201ea74fc9